skglm/solvers/prox_newton.py

```diff
@@ -242,7 +253,7 @@ def _descent_direction_s(X_data, X_indptr, X_indices, y, w_epoch,
 
     # see _descent_direction() comment
     past_grads = np.zeros(len(ws))
-    X_delta_w_ws = np.zeros(len(y))
+    X_delta_w_ws = np.zeros(Xw_epoch.shape[0])
     ws_intercept = np.append(ws, -1) if fit_intercept else ws
```

Comment:
A small hack that we used to implement Cox estimator was to pass in ``y`` as tuple ``(tm, s)``.
Doing so would break this function.